cmd/plugin/builder/inventory/inventory_plugin.go

```diff
@@ -0,0 +1,165 @@
+	}
+
+	for i := range pluginInventoryEntries {
+		err := inventoryUpdater(dbFile, pluginInventoryEntries[i])
+		if err != nil {
+			return err
+		}
+	}
+
+	// Publish the database to the remote repository
+	log.Info("Publishing plugin inventory database")
+	err = ipuo.ImgpkgOptions.PushImage(pluginInventoryDBImage, dbFile)
+	if err != nil {
+		return errors.Wrapf(err, "error while publishing inventory database to the repository as image: %q", pluginInventoryDBImage)
+	}
+	log.Infof("Successfully published plugin inventory database at: %q", pluginInventoryDBImage)
+
+	return nil
+}
+
+func (ipuo *InventoryPluginUpdateOptions) getPluginInventoryEntries() ([]*plugininventory.PluginInventoryEntry, error) {
+	pluginManifest, err := helpers.ReadPluginManifest(ipuo.ManifestFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var pluginInventoryEntries []*plugininventory.PluginInventoryEntry
+
+	for i := range pluginManifest.Plugins {
+		var pluginInventoryEntry *plugininventory.PluginInventoryEntry
+
+		for _, osArch := range cli.MinOSArch {
+			for _, version := range pluginManifest.Plugins[i].Versions {
+				pluginInventoryEntry, err = ipuo.updatePluginInventoryEntry(pluginInventoryEntry, pluginManifest.Plugins[i], osArch, version)
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+
+		pluginInventoryEntries = append(pluginInventoryEntries, pluginInventoryEntry)
+	}
+
+	return pluginInventoryEntries, nil
+}
+
+func (ipuo *InventoryPluginUpdateOptions) updatePluginInventoryEntry(pluginInventoryEntry *plugininventory.PluginInventoryEntry, plugin cli.Plugin, osArch cli.Arch, version string) (*plugininventory.PluginInventoryEntry, error) {
+	log.Infof("Validating plugin '%s_%s_%s_%s'", plugin.Name, plugin.Target, osArch.String(), version)
+
+	pluginImageBasePath := fmt.Sprintf("%s/%s/%s/%s/%s/%s:%s", ipuo.Vendor, ipuo.Publisher, osArch.OS(), osArch.Arch(), plugin.Target, plugin.Name, version)
+	pluginImage := fmt.Sprintf("%s/%s", ipuo.Repository, pluginImageBasePath)
+	digest, err := ipuo.ImgpkgOptions.GetFileDigestFromImage(pluginImage, cli.MakeArtifactName(plugin.Name, osArch))
+	if err != nil {
+		return nil, errors.Wrapf(err, "error while getting plugin binary digest from the image %q", pluginImage)
+	}
+
+	if pluginInventoryEntry == nil {
+		pluginInventoryEntry = &plugininventory.PluginInventoryEntry{
+			Name:        plugin.Name,
+			Target:      configtypes.Target(plugin.Target),
+			Description: plugin.Description,
+			Publisher:   ipuo.Publisher,
+			Vendor:      ipuo.Vendor,
+			Artifacts:   make(map[string]distribution.ArtifactList),
+			Hidden:      ipuo.DeactivatePlugins,
+		}
+	}
+	_, exists := pluginInventoryEntry.Artifacts[version]
+	if !exists {
+		pluginInventoryEntry.Artifacts[version] = make([]distribution.Artifact, 0)
+	}
+
+	artifact := distribution.Artifact{
+		OS:     osArch.OS(),
+		Arch:   osArch.Arch(),
+		Digest: digest,
+		URI:    pluginImageBasePath,
```

Comment:
Question: With central repository(OCI) we would be only using `Image`, is there a reason assigning the `pluginImageBasePath` to URI as well?

Reply:
Nope. Fixed.